subsys/bluetooth/host/hci_core.c

```diff
@@ -5372,6 +5375,26 @@ int bt_le_scan_stop(void)
 }
 #endif /* CONFIG_BT_OBSERVER */
 
+int bt_le_set_ch_map(u8_t ch_map[5])
+{
+	struct bt_hci_cp_le_set_host_chan_classif *cp;
```

Comment:
Should this function start with the following:
```
if (!IS_ENABLED(CONFIG_BT_CENTRAL)) {
        return -ENOTSUP;
}
````
since the core spec says the command should be used only if the host is capable of acting in master role?

Reply:
It depends, if we agree to keep the functionality of using bt_le_set_ch_map to set the advertiser channel map. Then we can wrap the issuing of the command in ifdef BT_CENTRAL.
If not then your suggestion would be best.